gen.py

```diff
@@ -67,10 +73,45 @@ def generate_single_tag(args: Any):
     repo = Repo.default()
     repo.fetch()
     version = args.version
-    _generate_tag(repo, version)
+    _generate_tag(repo, version, force=args.force)
     _generate_list_of_packages()
 
 
+def _nix_prefetch_git(url: str, rev: str) -> str:
+    return _run(f"nix-prefetch-git {url} --rev {rev} --quiet").decode("utf-8")
+
+
+def _prefetch_output_hashes(cargo_lock: str) -> str:
```

Comment:
Let's make it a generator yielding output hashes

Reply:
Clearly we think alike :smile: 

I actually had exactly that but felt that the final conversion to a string was a bit awkward simply stuck in the scope of the `generate_tag` method so I just made this output the final result. 

I'll add another outer function layer wrapping the generator that will build up the output iteratively, and maybe make the code a little more functional along the way.

Reply:
Actually, I realized that it's unlikely we'll have no matching packages, and the constructed default.nix will still be valid if it includes this
```nix
    outputHashes = {
    };
```

So I'll also simplify the logic and just have it yield the `"<pname>" = "<hash>"` strings